plugin/src/main/java/org/opensearch/ml/sdkclient/SdkClientFactory.java

```diff
@@ -110,6 +105,15 @@ public static SdkClient createSdkClient(Client client, NamedXContentRegistry xCo
         }
     }
 
+    private static void validateAwsParams(String clientType, String remoteMetadataEndpoint, String region, String serviceName) {
+        if (Strings.isBlank(remoteMetadataEndpoint) || Strings.isBlank(region)) {
+            throw new OpenSearchException(clientType + " client requires a metadata endpoint and region.");
+        }
+        if (!"es".equals(serviceName) && !"aoss".equals(serviceName)) {
```

Comment:
What if an opensource community wants to use some other service other than `es` or `aoss`? I think this part needs to be generic? Isn't that the whole point we moved this service name to a settings `REMOTE_METADATA_SERVICE_NAME_KEY`?

Reply:
These are specific to `case SdkClientSettings.AWS_OPENSEARCH_SERVICE:` and `case SdkClientSettings.AWS_DYNAMO_DB` for the `remoteMetadata` type.  Any other service or any other provider can be added to that configuration.

These two particular clients use the AWS SDK for connection and "es" and "aoss" are the only documented options right now, see https://opensearch.org/docs/latest/clients/java/#connecting-to-amazon-opensearch-service

Reply:
Also relevant: https://github.com/opensearch-project/data-prepper/blob/627ae9ffdb2de5dda84566e7b6a4f130f5005210/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchClientFactory.java#L54-L55

Reply:
> What if an opensource community wants to use some other service other than `es` or `aoss`? I think this part needs to be generic? Isn't that the whole point we moved this service name to a settings `REMOTE_METADATA_SERVICE_NAME_KEY`?

For a concrete example, let's say OCI wants to have a client connect to their service through the [OCI OpenSearch SDK](https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/javasdk.htm).   They would:
1. Define a new `SdkClientSettings.REMOTE_METADATA_TYPE`.  We already have "RemoteOpenSearch" (a cluster anywhere) and "AWSOpenSearchService" (AOS).  They'd add a new "OCISearchService" (or whatever name)
2. Instead of calling validation via `validateAWSParams()` which requires an endpoint, region, and one of two service names, they'd create a new `validateOCIParams()` checker method for whichever settings they needed to support methods on https://github.com/oracle/oci-java-sdk

Reply:
Thanks for detailed explanation. Makes sense, sounds good to me.